client/my-sites/themes/single-site-jetpack.jsx

```diff
@@ -120,6 +121,7 @@ const ConnectedSingleSiteJetpack = connectOptions( props => {
 				{ siteId && <QuerySitePlans siteId={ siteId } /> }
 				{ siteId && <QuerySitePurchases siteId={ siteId } /> }
 				<ThanksModal source={ 'list' } />
+				<AutoLoadingHomepageModal source={ 'list' } />
```

Comment:
I was wondering what this `source` argument is for. Is it just to pass into the `activateTheme` handler prop? If so could you not partially apply it to the `activateTheme` prop when you pass it in?

Reply:
`source` is the context where the theme is activated. This modal just [propagates this value](https://github.com/Automattic/wp-calypso/pull/39436/files#diff-dcbe1254be3df4a64783d3aa85867606R56) to the [activateTheme()](https://github.com/Automattic/wp-calypso/blob/master/client/state/themes/actions.js#L386) selector.

Reply:
I understand. This is why I was wondering why it needed to exist when you could partially apply the "known" argument `list` to `activateTheme` instead of passing the value in as a separate prop. 

Looks like you might be following an existing convention in Calypso, in which case I guess it's ok. Just seemed slightly redundant to me.